experimental/sdk/webpubsub-socketio-extension/src/SIO/components/web-pubsub-adapter.ts

```diff
@@ -49,9 +63,45 @@ export class WebPubSubAdapterInternal extends NativeInMemoryAdapter {
    * @param packet - the packet object
    * @param opts - the options
    */
-  public override broadcast(packet: Packet, opts: BroadcastOptions): void {
-    // TODO: Implement this method.
-    // For now, it is intentionally empty.
+  public override async broadcast(packet: Packet, opts: BroadcastOptions): Promise<void> {
+    debug(`broadcast packet ${JSON.stringify(packet)}`);
+    try {
+      const flags = opts.flags || {};
+      const packetOpts = {
+        preEncoded: true,
+        volatile: flags.volatile,
+        compress: flags.compress,
+      };
+
+      (packet as any).nsp = this.nsp.name;
+
+      const encodedPackets = (this as any)._encode(packet, packetOpts);
+
+      const oDataFilter = this._buildODataFilter(opts.rooms, opts.except);
+
+      debug(`broadcast encodedPackets = ${encodedPackets}, oDataFilter = ${oDataFilter}`);
+
+      // https://github.com/socketio/socket.io/blob/main/lib/client.ts#L230
+      if (encodedPackets.volatile) {
+        debug("volatile packet is discarded since the transport is not currently writable");
+        return;
+      }
+      const packets = Array.isArray(encodedPackets) ? encodedPackets : [encodedPackets];
+      for (let encodedPacket of packets) {
+        const sendOptions = { filter: oDataFilter };
+        if (typeof encodedPacket === "string") {
+          encodedPacket = "4" + encodedPacket.toString();
+          sendOptions["contentType"] = "text/plain";
+        } else {
+          sendOptions["contentType"] = "application/octet-stream";
+        }
+
+        await this.service.sendToAll(encodedPacket, sendOptions);
+      }
+    }
+    catch (error) {
+      console.error(error);
```

Comment:
is throw better? shall user be aware of the error? what is current socketio behavior when send fails?

Reply:
How native socket.io handle broadcast error:
[In-memory Adatper](https://github.com/socketio/socket.io-adapter/blob/main/lib/index.ts#L161): No error handling. No try-catch. User will be aware of error.
[Redis Adapter](https://github.com/socketio/socket.io-redis-adapter/blob/main/lib/cluster-adapter.ts#L298): For local broadcast, no error handling, User will be aware of error. For non-local broadcast, use try-catch and user won't be aware 
 of error
[Redis Stream Adapter](https://github.com/socketio/socket.io-redis-streams-adapter/blob/main/lib/cluster-adapter.ts#L294) same as redis adapter.

Maybe follow in-memory adapter is better? No any error handling and let user handle all errors. Is this way ok? Cc @zackliu